custom-metrics-stackdriver-adapter/README.md

```diff
@@ -0,0 +1,127 @@
+     - For a GCE instance `<my_instance>` use following command:
+       ```
+       gcloud compute instances describe <my_instance>
+       ```
+       `https://www.googleapis.com/auth/monitoring` should be listed in the
+       `scopes` section.
+
+
+     To configure set scopes manually, you can use:
+     - `--scopes` flag if you are using `gcloud container clusters create`
+       command, see [gcloud
+       documentation](https://cloud.google.com/sdk/gcloud/reference/container/clusters/create).
+     - Environment variable `NODE_SCOPES` if you are using [kube-up.sh script].
+       It is enabled by default.
+     - To set scopes in existing clusters you can use `gcloud beta compute
+       instances set-scopes` command, see [gcloud
+       documentation](https://cloud.google.com/sdk/gcloud/reference/beta/compute/instances/set-scopes).
+
+1. Start *Custom Metrics - Stackdriver Adapter*.
+
+```sh
+kubectl create -f
+https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/adapter-beta.yaml
+```
+
+### Export metrics to Stackdriver
+
+To learn how to create your custom metric and write your data to Stackdriver,
+follow [Stackdriver custom metrics documentation]. You can also follow
+[Prometheus to Stackdriver documentation] to export metrics exposed by your pods
+in Prometheus format.
+
+The name of your metric must start with custom.googleapis.com/ prefix followed
+by a simple name, as defined in [custom metric naming rules].
+For purpose of this guide, we will use metric `custom.googleapis.com/qps`.
+
+1. Define your custom metric. Your metric descriptor needs to meet following requirements:
+   * `metricKind = GAUGE`
+   * `metricType = DOUBLE` or `INT64`
+1. Export metric from your application. The metric has to be associated with a
+   specific pod and meet folowing requirements:
+   * `resource_type = "gke_container"` (See monitored resources documentation)
+   * Following resource labels are set to correct values:
+     - `pod_id` - set this to pod UID obtained via downward API. Example configuration that passes POD_ID to your application as a flag:
+
+     ```yaml
+     apiVersion: v1
+     kind: Pod
+     metadata:
+       name: my-pod
+     spec:
+       containers:
+       - image: <my-image>
+         command:
+         - my-app --pod_id=$(POD_ID)
+         env:
+         - name: POD_ID
+           valueFrom:
+             fieldRef:
+               fieldPath: metadata.uid
+     ```
+
+     - `container_name = ""`
+     - `project_id`, `zone`, `cluster_name` - can be obtained by your application from
+        [metadata server]. You can use Google Cloud compute metadata client to get
+        these values, example in Go:
+
+     ```go
+     import gce "cloud.google.com/go/compute/metadata"
+
+     project_id, err := gce.ProjectID()
+     zone, err := gce.Zone()
+     cluster_name, err := gce.InstanceAttributeValue("cluster-name")
+     ```
+
+#### Example
```

Comment:
If we can find a smaller example it might be a good idea to just put it inline with the Go snippet above.

Reply:
The linked example is pretty minimal. I have added two more snippets which I think are sufficient to export the metric (still leaving a link to example app to provide some code that can be compiled and tried out right away).